lib/facility.rb

```diff
@@ -1,14 +1,59 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
-  end
-
-  def add_services(service)
-    @services << service
-  end
+
+    attr_reader :name,
+                :address,
+                :phone,
+                :services,
+                :collected_fees,
+                :registered_vehicles
+
+    def initialize(facility_hash)
+        @name = facility_hash[:name]
+        @address = facility_hash[:address]
+        @phone = facility_hash[:phone]
+        @services = []
+        @collected_fees = 0
+        @registered_vehicles = []
+    end
+
+    def add_service(service)
+        @services << service
+    end
+
+    def register_vehicle(vehicle)
```

Comment:
Excellent work structuring these methods to be clean and readable. Despite their complexity, you're making great use of helper methods to  break things up. Looks awesome!